great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -52,69 +54,63 @@ class SQLDatasourceWarning(UserWarning):
     pass
 
 
-@pydantic_dc.dataclass(frozen=True)
-class ColumnSplitter:
+class _ColumnSplitter(ExperimentalBaseModel):
     column_name: str
     method_name: str
 
+    @property
+    def param_names(self) -> List[str]:
+        raise NotImplementedError
+
+    def splitter_method_kwargs(self) -> Dict[str, Any]:
```

Comment:
@billdirks What would be a helpful docstring explaining this interface method for developers needing to extend it?  Thanks.

Reply:
Done. Also, please see the note above as the this not currently being a point of extensibility.